chia/farmer/farmer_api.py

```diff
@@ -256,17 +270,20 @@ async def new_proof_of_space(
 
                 # The plot key is 2/2 so we need the harvester's half of the signature
                 m_to_sign = payload.get_hash()
-                m_src_data: Optional[List[SignatureRequestSourceData]] = None
+                m_src_data: Optional[List[Optional[SignatureRequestSourceData]]] = None
```

Comment:
I believe this is because there may not be a `FoliageTransactionBlock` sent over from the `FullNode` if the `UnfinishedBlock`'s `is_transaction_block()` returns `False`, so we have nothing to place there as the source data in that case.